src/osd/osd_types.cc

```diff
@@ -3123,6 +3123,10 @@ bool pg_interval_t::check_new_interval(
       if (*p != CRUSH_ITEM_NONE)
 	++num_acting;
 
+    if (!lastmap->get_pools().count(pgid.pool())) {
+      i.maybe_went_rw = true;
+      return true;
+    }
```

Comment:
Hmm, the fact that the caller is required to pass in old_up (etc) suggests that assert(lastmap->get_pools().count(pgid.pool()) is a precondition of this function.  There appear to be 3 callers.  Two are variants of build_past_intervals and they get their bounding epochs from _calc_past_interval_range.  _calc_past_interval_range gets its lower bound from last_epoch_clean or epoch_created either of which must be an epoch where the pool exists.  The third is in start_peering_interval which can only be called during an interval in which the pg/pool exists.

In summary, I'm pretty sure that's a precondition for the method, make it an assert.